### PR TITLE
added DataContext and reducer

### PR DESCRIPTION
**Pull request by Rohanmond**
Added initial data context and reducer and got data from initial get

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,8 +1,14 @@
 import { Route, Routes } from 'react-router-dom';
+import { useEffect } from 'react';
 import './App.css';
 import { MockAPI } from './components';
+import { useData } from './contexts';
 
 function App() {
+  const { state } = useData();
+  useEffect(() => {
+    console.log(state);
```
Comment (rutvikpumak) [inline src/App.js:10]: remove console log

File: src/App.js [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
+  }, [state]);
```
Comment (znjs) [inline src/App.js:11]: Can remove the `console.log`'s

File: src/Services/services.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import axios from 'axios';
+
+export const GetAllVideos = async () => {
+  return axios.get('/api/videos');
+};
+
+export const GetAllCategories = async () => {
+  return axios.get('/api/categories');
+};
```
Comment (znjs) [inline src/Services/services.js:9]: Returning a promise not value
Check once if this is what is intended

**Review by znjs — APPROVED** [review]
Good folder structure
good naming conventions used

**Review by rutvikpumak — APPROVED** [review]
Code looks clean and well structured 
Good to merge Just remove console

File: src/App.js (hunk 2 above)
Comment (Rohanmond) [reply src/App.js:11]: I put console logs to check the data, in the next I will remove thsese

File: src/Services/services.js (hunk 3 above)
Comment (Rohanmond) [reply src/Services/services.js:9]: it is intentional, I'm returing a promise only here

File: src/App.js (hunk 1 above)
Comment (Rohanmond) [reply src/App.js:10]: sure

**Pull request merged**